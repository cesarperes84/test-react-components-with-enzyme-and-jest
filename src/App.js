import React, { Component } from 'react';
// import logo from './logo.svg';
// import './App.css';

const Test = () => <div>Testing</div>;

const Title = ({text}) => <div>{text}</div>;

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>;
  }
};

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      input: '',
      on: false,
    }
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          {/* <img src={logo} className="App-logo" alt="logo" /> */}
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Title text="Some title" />
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <p className="button-state">{this.state.on ? 'Yes!' : 'No!'}</p>
        <button onClick={() => this.setState({on: true})}></button>
        <h2>{this.state.input}</h2>
        <input onChange={(e) => this.setState({input: e.currentTarget.value})} type="text" />
        <ul>
          <li></li>
          <li></li>
          <li></li>
        </ul>
        <Test />
      </div>
    );
  }
}

export default App;
