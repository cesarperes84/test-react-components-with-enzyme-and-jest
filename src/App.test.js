import React from 'react';
import ReactDOM from 'react-dom';
import App, { Link } from './App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import enzymeToJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  const wrapper = shallow(<App />);
  // const wrapper = shallow(<App />, {context: {}, disableLifecycleMethods: bool});
  it('should contain 1 p element', () => {
    // console.log(wrapper.debug());
    // expect(wrapper.find('p').length).toBe(1);
    expect(wrapper.find('.App-intro').exists()).toBe(true);
  });
  it('should contain ul element with lis', () => {
    expect(wrapper.find('ul').children().length).toBe(3);
  });
  it('should contain h1 element with correct text', () => {
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(enzymeToJson(tree)).toMatchSnapshot();
  });
});

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe('<Link />', () => {
  it('link component accepts address prop', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.instance().props.address).toBe('www.google.com');
  });
  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.props().href).toBe('www.google.com');
  });
  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull();
  });
});
