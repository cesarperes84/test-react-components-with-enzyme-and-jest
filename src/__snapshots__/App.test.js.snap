// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <Title
      text="Some title"
    >
      <div>
        Some title
      </div>
    </Title>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <ul>
      <li />
      <li />
      <li />
    </ul>
    <Test>
      <div>
        Testing
      </div>
    </Test>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <Title
    text="Some title"
  />
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <ul>
    <li />
    <li />
    <li />
  </ul>
  <Test />
</div>
`;
